nubis/puppet/files/nat/confd/templates/iptables.tmpl

```diff
@@ -15,617 +6,198 @@ resources { 'firewall':
-    destination => $vpc_cidr,
-    proto       => 'tcp',
-    dport       => [ '8300', '8301', '8302', '8400', '8500', '8600' ],
-    sport       => '1025-65535',
-    action      => 'accept',
-}
-
-firewall { '007 allow inbound tcp consul eth0':
-    chain     => 'INPUT',
-    iniface   => 'eth0',
-    source    => $vpc_cidr,
-    proto     => 'tcp',
-    tcp_flags => '! SYN,RST,ACK,FIN SYN',
-    sport     => [ '8300', '8301', '8302', '8400', '8500', '8600' ],
-    dport     => '1025-65535',
-    action    => 'accept',
-}
-
-# consul outbound port 8301 tcp
-firewall { '007 allow inbound tcp port 8301 eth1':
-    chain   => 'INPUT',
-    iniface => 'eth1',
-    source  => $vpc_cidr,
-    proto   => 'tcp',
-    dport   => '8301',
-    sport   => '1025-65535',
-    action  => 'accept',
-}
-
-firewall { '007 allow outbound tcp port 8301 eth1':
-    chain       => 'OUTPUT',
-    outiface    => 'eth1',
-    destination => $vpc_cidr,
-    proto       => 'tcp',
-    sport       => '8301',
-    dport       => '1025-65535',
-    action      => 'accept',
+# Allow all traffic thatis part of already known flows
+firewall { '000 accept established flows':
+  chain   => 'INPUT',
+  proto   => 'all',
+  state   => [ 'ESTABLISHED','RELATED' ],
+  action  => 'accept',
+}
+
+# Allow all loopback in
+firewall { '001 loopback in':
+  chain     => 'INPUT',
+  proto     => 'all',
+  state     => [ 'NEW' ],
+  iniface   => 'lo',
+  action    => 'accept',
+}
+
+# Allow DHCP in
+firewall { '002 dhcp in':
+  chain     => 'INPUT',
+  proto     => 'udp',
+  dport     => [ 'bootpc','bootps' ],
+  action    => 'accept',
+}
+
+# Allow all ICMP in
+firewall { '003 ICMP in':
+  chain     => 'INPUT',
+  proto     => 'icmp',
+  state     => [ 'NEW' ],
+  action    => 'accept',
+}
+
+# Allow ssh IN internally
+firewall { '004 SSH in':
+  chain     => 'INPUT',
+  proto     => 'tcp',
+  dport     => 'ssh',
+  state     => [ 'NEW' ],
+#: For convenience, allow ssh from anywhere, rely on security groups to filter that out
+#  iniface   => 'eth1',
+  action    => 'accept',
+}
+
+# Allow Squid/tcp IN internally
+firewall { '005 Squid/tcp in':
+  chain     => 'INPUT',
+  proto     => 'tcp',
+  dport     => '3128',
+  state     => [ 'NEW' ],
+  iniface   => 'eth1',
+  action    => 'accept',
+}
+
+# Allow Consul/tcp IN internally
+firewall { '006 Consul/tcp in':
+  chain     => 'INPUT',
+  proto     => 'tcp',
+  dport     => '8300-8302',
+  state     => [ 'NEW' ],
+  iniface   => 'eth1',
+  action    => 'accept',
+}
+
+# Allow Consul/udp IN internally
+firewall { '007 Consul/udp in':
+  chain     => 'INPUT',
+  proto     => 'udp',
+  dport     => '8300-8302',
+  state     => [ 'NEW' ],
+  iniface   => 'eth1',
+  action    => 'accept',
+}
+
+# Allow Consul/udp IN internally
+firewall { '008 node monitoring IN':
+  chain     => 'INPUT',
+  proto     => 'tcp',
+  dport     => '9100-9110',
+  state     => [ 'NEW' ],
+  iniface   => 'eth1',
+  action    => 'accept',
 }
 
```

Comment:
That's because the firewall as built before wasn't stateful, so the trace you are showing (ACK SYN) is a return packet being denied.

In my ruleset, the INPUT ESTABLISHED rule would have allowed it back in.